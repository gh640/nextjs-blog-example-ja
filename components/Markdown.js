/**
 * 共通のマークダウンレンダラー
 */
import * as React from "react"
import ReactMarkdown from "react-markdown"
import rehypeRaw from "rehype-raw"
import prism from "prismjs"

import "prismjs/themes/prism.css"
import "prismjs/themes/prism-twilight.css"

const Markdown = ({ children }) => {
  React.useEffect(() => {
    prism.highlightAll()
  })

  // Markdown 中の HTML を許容するために `rehypeRaw` を利用する
  return (
    <ReactMarkdown rehypePlugins={[rehypeRaw]}>{children}</ReactMarkdown>
  )
}

export default Markdown
