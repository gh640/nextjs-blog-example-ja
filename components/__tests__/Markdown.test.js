import renderer from "react-test-renderer"

import Markdown from "../Markdown.js"

jest.mock("next/link", () => {
	return ({ children, ...attrs }) => {
		return <a {...attrs}>{children}</a>
	}
})

describe("Markdown", () => {
	it("renders collectly with simple markdown", () => {
		const tree = renderer
			.create(<Markdown>{`
# 見出し 1

# 見出し 2

- これは
- リスト
- です

テーブル:

| 西 | 東 |
| --- | --- |
| 西山 | 東山 |

			`}</Markdown>)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it("renders collectly with html", () => {
		const tree = renderer
			.create(<Markdown>{`
<h3>テーブルサンプル</h3>
<div>
	<table>
		<thead>
			<tr>
				<th>カラム 1</th>
				<th>カラム 2</th>
			</tr>
		</thead>
		<tbody>
			<tr>
				<td>1.1</td>
				<th>1.2</th>
			</tr>
			<tr>
				<td>2.1</td>
				<th>2.2</th>
			</tr>
		</tbody>
	</table>
</div>
			`}</Markdown>)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

})
