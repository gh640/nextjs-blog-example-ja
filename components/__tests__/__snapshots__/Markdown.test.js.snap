// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown renders collectly with html 1`] = `
Array [
  <h3>
    テーブルサンプル
  </h3>,
  "
",
  <div>
    
	
    <table>
      <thead>
        <tr>
          <th>
            カラム 1
          </th>
          <th>
            カラム 2
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1.1
          </td>
          <th>
            1.2
          </th>
        </tr>
        <tr>
          <td>
            2.1
          </td>
          <th>
            2.2
          </th>
        </tr>
      </tbody>
    </table>
    

  </div>,
]
`;

exports[`Markdown renders collectly with simple markdown 1`] = `
Array [
  <h1>
    見出し 1
  </h1>,
  "
",
  <h1>
    見出し 2
  </h1>,
  "
",
  <ul>
    

    <li>
      これは
    </li>
    

    <li>
      リスト
    </li>
    

    <li>
      です
    </li>
    

  </ul>,
]
`;
